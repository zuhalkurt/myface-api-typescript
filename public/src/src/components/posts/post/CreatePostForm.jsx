import React, {useState} from "react";
import { createPost } from "../../../clients/myFaceClients";

export function CreatePostForm() {
    const [message, setMessage] = useState("");
    const [imageUrl, setImageUrl] = useState("");

    return <form onSubmit={e => {
        e.preventDefault();
        createPost(message, imageUrl);
    }}>
        <p>Message: {message}, ImageUrl:{imageUrl}</p>
        <label for="message">Message:</label>
        <input
            type="text"
            id="message"
            name="message"
            onChange={e=> setMessage(e.target.value)}        
        />
        <label for="imageUrl">Image URL:</label>
        <input
            type="text"
            id="imageUrl"
            name="imageUrl"
            onChange={e => setImageUrl(e.target.value)}
        />
        <input type="submit" value="Submit" />
    </form>
}