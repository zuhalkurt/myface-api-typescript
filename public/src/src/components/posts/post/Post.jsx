import React from "react";
import "./Post.scss";


export function Post({ post }) {
    return <div className="post">
        <h2>{post.message}</h2>
        <img src={post.imageUrl} alt={post.message} />
        {
           post.postedBy !== undefined
           ? <p>Posted by {post.postedBy.name} on {post.createdAt}</p>
           : <p>{post.createdAt}</p>
        }
        
    </div>
}