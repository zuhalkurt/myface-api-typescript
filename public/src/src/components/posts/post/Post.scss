@import "../../../constants.scss";

.post {
    background-color: $darkerBackgroundColour;
    display: grid;
    grid-template-columns: 40vw 1fr;
    grid-template-rows: 0.6fr 0.4fr;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
    height: 40vw;

    img {
        grid-column: 1 / 2;
        grid-row: 1 / 3;
        width: 40vw;
        height: 40vw;
    }

    h2 {
        grid-column: 2 / 3;
        grid-row: 1 /2;
        font-size: $mobile-subtitle-font-size;
        line-height: $mobile-subtitle-line-height;
        margin: 0;
        align-self: center;
    }
    p {
        margin: 0;
    }
}