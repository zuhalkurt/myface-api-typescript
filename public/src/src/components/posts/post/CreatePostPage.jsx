import React, {useState} from "react";
import { CreatePostForm } from "./CreatePostForm";

export function CreatePostPage() {
    return <main>
        <CreatePostForm/>
    </main>
}
