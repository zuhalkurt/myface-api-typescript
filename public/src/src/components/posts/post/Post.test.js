import React from 'react';
import renderer from "react-test-renderer";
import { render, screen} from '@testing-library/react';
import { Post } from "../post/Post";


test('renders post message', () => {
    const post = {
      message: "Post message",
      imageUrl: "Image Url",
      createdAt: "2022/04/05 11:00",
      postedBy: {
        name: "User name"
      },
    };

    render(<Post post={post} />);
    const messageElement = screen.getByText(/post message/i);
    expect(messageElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  const post = {
    message:"Post message",
    imageUrl: "Image Url",
    createAt: "2022/04/05 11:00",
    postedBy: {
      name:"User name",
    },
  };

  const tree = renderer
    .create(<Post post={post}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});



