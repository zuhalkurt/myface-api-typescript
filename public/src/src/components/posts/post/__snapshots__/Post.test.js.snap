// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="post"
>
  <h2>
    Post message
  </h2>
  <img
    alt="Post message"
    src="Image Url"
  />
  <p>
    Posted by 
    User name
     on 
  </p>
</div>
`;
